src/text.c

```diff
@@ -381,29 +429,53 @@ bool32 AddTextPrinter(struct TextPrinterTemplate *printerTemplate, u8 speed, voi
 void RunTextPrinters(void)
 {
     int i;
+    bool32 isInstantText = IsPlayerTextSpeedInstant(); // Force correct result. This is dumb, Revo knows.
+    u32 textRepeats = GetPlayerTextSpeedModifier();
 
-    if (!gDisableTextPrinters)
+    do
     {
-        for (i = 0; i < WINDOWS_MAX; ++i)
+        u32 numEmpty = 0;
+        if (!gDisableTextPrinters)
         {
```

Comment:
```suggestion
        if (!gDisableTextPrinters)
        {
```
By inverting this you will save one depth which will make it a bit more readable.